glean-core/csharp/Glean/Dispatchers.cs

```diff
@@ -106,164 +90,101 @@ internal static Task LaunchAPI(Action action)
 
             if (QueueInitialTasks)
             {
-                // If we are queuing tasks, typically before Glean has been
-                // initialized, then we should just add the created Task to the
-                // taskQueue.
+                // If we are queuing, typically before Glean has been
+                // initialized, then we should just add the action to
+                // the queue.
                 AddActionToQueue(action);
             }
             else
             {
                 if (!TestingMode)
                 {
-                    // If we are not queuing initial tasks, we can go ahead and
-                    // execute the task asynchronously on the ExclusiveScheduler                    
-                    task = Task.Factory.StartNew(
-                        () =>
-                        {
+                    // If we are not queuing we can go ahead and execute the
+                    // task asynchronously                    
+                    task = factory.StartNew(() =>
+                    {
                             // In order to prevent tasks from causing exceptions
                             // we wrap the action invocation in try/catch
                             try
-                            {
-                                action.Invoke();
-                            }
-                            catch (Exception)
-                            {
+                        {
+                            action.Invoke();
+                        }
+                        catch (Exception)
+                        {
                                 //TODO Exception eaten by Glean
                             }
-                        },
-                        cancellationToken,
-                        TaskCreationOptions.None,
-                        taskSchedulerPair.ExclusiveScheduler);
+                    });
                 }
                 else
                 {
-                    // If we are in testing mode, then go ahead and invoke the
-                    // action to ensure synchronous execution.
-                    action.Invoke();
+                    // If we are in testing mode, then go ahead and await
+                    // the task to ensure synchronous execution.
+                    factory.StartNew(action).Wait();
```

Comment:
Why does this have to change? Can't this be just `action.Invoke()`?

Reply:
It can be just `Invoke()` here, good call.